File: 2695-find-score-of-an-array-after-marking-all-elements/2695-find-score-of-an-array-after-marking-all-elements.go
func findScore(nums []int) int64 {
    var ans int64 = 0
    n := len(nums)
    for i := 0; i < n; i += 2 {
        i0 := i
        for i + 1 < n && nums[i] > nums[i+1] {
            i++
        }
        for j := i; j >= i0; j -= 2 {
            ans += int64(nums[j])
        }
    }
    return ans
}

File: 2695-find-score-of-an-array-after-marking-all-elements/2695-find-score-of-an-array-after-marking-all-elements.c
long long findScore(int* nums, int numsSize) {
    long long ans = 0;
    for (int i = 0, n = numsSize; i < n; i += 2) {
        int i0 = i;
        while (i + 1 < n && nums[i] > nums[i + 1])
            ++i;
        for (int j = i; j >= i0; j -= 2)
            ans += nums[j];
    }
    return ans;
}